lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -182,50 +182,66 @@
+            - List of data disks.
         version_added: "2.4"
         suboptions:
             lun:
                 description:
-                    - The logical unit number for data disk
+                    - The logical unit number for data disk.
                 default: 0
                 version_added: "2.4"
             disk_size_gb:
                 description:
-                    - The initial disk size in GB for blank data disks
+                    - The initial disk size in GB for blank data disks.
                 version_added: "2.4"
             managed_disk_type:
                 description:
-                    - Managed data disk type
+                    - Managed data disk type.
                 choices:
                     - Standard_LRS
                     - StandardSSD_LRS
+                    - UltraSSD_LRS
                     - Premium_LRS
                 version_added: "2.4"
+            managed_disk_id:
+                description:
+                    - Managed data disk id.
+                    - Only used when OS disk created with managed disk.
+                version_added: "2.9"
             storage_account_name:
                 description:
-                    - Name of an existing storage account that supports creation of VHD blobs. If not specified for a new VM,
-                      a new storage account named <vm name>01 will be created using storage type 'Standard_LRS'.
+                    - Name of an existing storage account that supports creation of VHD blobs.
+                    - If not specified for a new data disk, a new storage account named <vm name>01 will be created using storage type C(Standard_LRS).
                 version_added: "2.4"
             storage_container_name:
                 description:
-                    - Name of the container to use within the storage account to store VHD blobs. If no name is specified a
-                      default container will created.
+                    - Name of the container to use within the storage account to store VHD blobs.
+                    - If not specified, a default container will created.
                 default: vhds
                 version_added: "2.4"
             storage_blob_name:
                 description:
-                    - Name fo the storage blob used to hold the VM's OS disk image. If no name is provided, defaults to
-                      the VM name + '.vhd'. If you provide a name, it must end with '.vhd'
+                    - Name of the storage blob used to hold the OS disk image of the VM.
+                    - Must end with '.vhd'.
+                    - If not specified, defaults to the VM name + '.vhd'.
                 version_added: "2.4"
             caching:
                 description:
-                    - Type of data disk caching.
+                    - Specifies the caching requirements.
                 choices:
+                    - empty
```

Comment:
Probably don't want to add this multi-case, multi-form support for all these options... Just use the CamelCase form that's currently used through the rest of the module- if someone wants to address the camel/snake thing in a bigger way throughout the module later and deprecate the CamelCase form, that should probably be its own PR.